server/src/test/java/org/elasticsearch/action/support/SubscribableListenerTests.java

```diff
@@ -607,6 +610,32 @@ public void testAndThenAcceptFailure() {
         assertComplete(chainedListener, "simulated");
     }
 
+    public void testRejection() {
```

Comment:
```suggestion
    public void testExecutorExecutionErrorPropagation() {
```

Reply:
Not an `Error` (and `ExecutorExecution` is kinda awkward), proposing `testRejectedExecutionThreading` instead, see d78df289017.